src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.Binding.cs

```diff
@@ -190,13 +216,13 @@ internal partial class CommonReferenceManager<TCompilation, TAssemblySymbol>
                             continue;
                         }
 
-                        var peReference = resolver.ResolveMissingAssembly(binding.ReferenceIdentity);
-                        if (peReference == null)
+                        var resolvedReference = resolver.ResolveMissingAssembly(requestingReference, binding.ReferenceIdentity);
```

Comment:
Filed issue: https://github.com/dotnet/roslyn/issues/5923. Will see if it makes thru ask mode. The scenario that would be enabled isn't so common.